### PR TITLE
Remove code that was inserting newline after = in a chained call

### PR DESCRIPTION
**Pull request by shashachu**
* Fixes https://github.com/pinterest/ktlint/issues/368 and https://github.com/pinterest/ktlint/issues/380
* Kotlin styleguide: https://kotlinlang.org/docs/reference/coding-conventions.html#chained-call-wrapping
  * Says 'The first call in the chain usually should have a line break before it, but it's OK to omit it if the code makes more sense that way.'

### CODE REVIEW

**Comment by shashachu** [comment]
@vanniktech thoughts on this one? I just removed the code that was inserting the newline after the `=` because it felt more like a bug than a feature.

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-dot-qualified-expression-expected.kt.spec [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class A {
-    private fun getImplementationVersion() =
-        javaClass.`package`.implementationVersion
-            ?: javaClass.getResourceAsStream("/META-INF/MANIFEST.MF")
-                ?.let { stream ->
-                    Manifest(stream).mainAttributes.getValue("Implementation-Version")
-                }
+    private fun getImplementationVersion() = javaClass.`package`.implementationVersion
```
Comment (JLLeitschuh) [inline ktlint-ruleset-experimental/src/test/resources/spec/indent/format-dot-qualified-expression-expected.kt.spec]: For functions like this, I actually think I like that Ktlint forces you to put your function body starting on a new line. It reads better.

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-raw-string-trim-indent-expected.kt.spec [hunk 2]
```diff
@@ -87,11 +87,10 @@ fun f() {
...
 class C {
-    val CONFIG_COMPACT =
-        """
-        {
-        }
-        """.trimIndent()
+    val CONFIG_COMPACT = """
+    {
+    }
+    """.trimIndent()
```
Comment (JLLeitschuh) [inline ktlint-ruleset-experimental/src/test/resources/spec/indent/format-raw-string-trim-indent-expected.kt.spec]: This kinda makes sense and kinda doesn't.
The consistent requirement on the `"""` being lined up would actually be better.

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-dot-qualified-expression-expected.kt.spec (hunk 1 above)
Comment (shashachu) [reply ktlint-ruleset-experimental/src/test/resources/spec/indent/format-dot-qualified-expression-expected.kt.spec]: Yeah in this case I'm sort of +/- on the change. Really the case that seems broken is:
```
val foo = someClass
    .aFunction()
    .anotherFunction()
```

becomes
```
val foo = 
    someClass
        .aFunction()
        .anotherFunction()
```

So maybe this is too heavy a hammer.

Comment (shashachu) [reply ktlint-ruleset-experimental/src/test/resources/spec/indent/format-dot-qualified-expression-expected.kt.spec]: @JLLeitschuh one thing to remember is that with this PR, ktlint won't disallow the code as it was originally formatted, it just won't force it.

**Comment by shashachu** [comment]
I re-introduced the rule for raw strings but removed the rest. I also added a test verifying the linebreak after = in chained calls is still allowed, just not forced.

File: ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/IndentationRuleTest.kt [hunk 3]
```diff
@@ -283,4 +283,22 @@ class IndentationRuleTest {
         assertThat(IndentationRule().format(ktScript, mapOf("indent_size" to "2")))
             .isEqualTo("fun main() {\n  return 0\n}")
     }
+
+    @Test
+    fun testLintNewlineAfterEqALlowed() {
```
Comment (NightlyNexus) [inline ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/IndentationRuleTest.kt]: accidentally capitalized L? `Allowed`

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec [hunk 4]
```diff
@@ -5,21 +5,19 @@ class C {
 }
 
 fun f() {
-    val x =
-        "a" +
-            "b2"
+    val x = "a" +
```
Comment (NightlyNexus) [inline ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec]: i don't know if this better...

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec [hunk 5]
```diff
@@ -5,21 +5,19 @@ class C {
...
 
-    val x =
-        paths.flatMap { dir ->
-            "hello"
-        } + f0(
-            "there"
-        ) + f1(
-            "sssss"
-        )
+    val x = paths.flatMap { dir ->
```
Comment (NightlyNexus) [inline ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec]: this does seem slightly better, though

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-property-accessor-expected.kt.spec [hunk 6]
```diff
@@ -1,9 +1,8 @@
 class C {
 
     private val Any.className
-        get() =
-            this.javaClass.name
-                .fn()
+        get() = this.javaClass.name
```
Comment (NightlyNexus) [inline ktlint-ruleset-experimental/src/test/resources/spec/indent/format-property-accessor-expected.kt.spec]: am also unsure here.

File: ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/IndentationRuleTest.kt (hunk 3 above)
Comment (shashachu) [reply ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/IndentationRuleTest.kt]: fixed.

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec (hunk 4 above)
Comment (shashachu) [reply ktlint-ruleset-experimental/src/test/resources/spec/indent/format-eq-expected.kt.spec]: really? it seems better to me. but the previous formatting is still allowed, just not forced.

File: ktlint-ruleset-experimental/src/test/resources/spec/indent/format-property-accessor-expected.kt.spec (hunk 6 above)
Comment (shashachu) [reply ktlint-ruleset-experimental/src/test/resources/spec/indent/format-property-accessor-expected.kt.spec]: both look bad to me, but the extra newline irks me. maybe it's just me. :)

**Pull request merged**